### PR TITLE
fix error "deque index out of range"

### LINKED ISSUE #1829: worker pod crashes when get task
```
2020-03-12 22:03:32.300671: W tensorflow/core/kernels/data/generator_dataset_op.cc:103] Error occurred when finalizing GeneratorDataset iterator: Cancelled: Operation was cancelled
Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/elasticdl/python/worker/main.py", line 44, in <module>
    main()
  File "/elasticdl/python/worker/main.py", line 40, in main
    worker.run()
  File "/elasticdl/python/worker/worker.py", line 1144, in run
    self._train_and_evaluate()
  File "/elasticdl/python/worker/worker.py", line 1074, in _train_and_evaluate
    self._minibatch_size, err_msg
  File "/elasticdl/python/worker/task_data_service.py", line 86, in report_record_done
    task = self._pending_tasks[0]
IndexError: deque index out of range
```

### PR DESCRIPTION
**Pull request by QiJune**
Fix #1829

### CODE REVIEW

File: elasticdl/python/worker/task_data_service.py [hunk 1]
```diff
@@ -83,6 +83,8 @@ def report_record_done(self, count, err_msg=""):
         if err_msg:
             self._failed_record_count += count
 
+        if not self._pending_tasks:
```
Comment (skydoorkai) [inline elasticdl/python/worker/task_data_service.py:87]: When will this happen that worker reports record done but there are no pending tasks?

File: elasticdl/python/worker/task_data_service.py [hunk 2]
```diff
@@ -83,6 +83,8 @@ def report_record_done(self, count, err_msg=""):
...
+            return False
         task = self._pending_tasks[0]
```
Comment (skydoorkai) [inline elasticdl/python/worker/task_data_service.py:89]: Try to add a lock for self._pending_tasks[0] access.

Comment (QiJune) [reply elasticdl/python/worker/task_data_service.py:89]: @skydoorkai I add a lock here, but the error still exists.

File: elasticdl/python/worker/task_data_service.py (hunk 1 above)
Comment (QiJune) [reply elasticdl/python/worker/task_data_service.py:87]: Add a workaround comment here. I will debug it later.

**Review by skydoorkai — APPROVED** [review]

**Pull request merged**